### DISCUSSION TITLE
renovate-config-validator: Apply --strict option to a file set in RENOVATE_CONFIG_FILE

### DISCUSSION THREAD

**Discussion by massongit** [renovatebot/renovate#23878] [Suggest an Idea]
### Tell us more.

In renovate-config-validator, I want to apply `--strict` option to a file set in `RENOVATE_CONFIG_FILE`.
Specifically, it is assumed to validate `default.json`.

**Comment by viceice** [comment]
duplicate of #22026

**Reply by massongit** [reply]
#23677, which is a PR that solves #22026, adds `--strict` option that is applied to `renovate.json` etc.
However, it is not applied to a file set in `RENOVATE_CONFIG_FILE`.

**Reply by viceice** [reply]
it should work for the ENV var too

https://github.com/renovatebot/renovate/blob/main/lib/config-validator.ts#L137

**Reply by massongit** [reply]
https://github.com/dev-hato/renovate-config/blob/ff9836e13dfa4f587b98401aca13b400e1823e26/package.json#L5
https://github.com/dev-hato/renovate-config/blob/ff9836e13dfa4f587b98401aca13b400e1823e26/.github/workflows/renovate-config-validator.yml#L20-L22

I ran the above with `--strict` option and `RENOVATE_CONFIG_FILE` set.
However, as shown below, it does not terminate abnormally even though migration is required.

https://github.com/dev-hato/renovate-config/actions/runs/5877736031/job/15938407741

```
 WARN: Config needs migrating
       "originalConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["config:base", ":timezone(Asia/Tokyo)"],
         "automerge": true,
         "platformAutomerge": true,
         "prConcurrentLimit": 2,
         "packageRules": [{"matchPackageNames": ["npm"], "enabled": false}],
         "lockFileMaintenance": {"enabled": true}
       },
       "migratedConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["config:recommended", ":timezone(Asia/Tokyo)"],
         "automerge": true,
         "platformAutomerge": true,
         "prConcurrentLimit": 2,
         "packageRules": [{"matchPackageNames": ["npm"], "enabled": false}],
         "lockFileMaintenance": {"enabled": true}
       }
 INFO: Validating default.json
```

https://github.com/renovatebot/renovate/blob/0d0ab69cfef1098d7b4e5877fcf5f0bce3172056/lib/workers/global/config/parse/file.ts#L83-L85


Since the config after migration is returned above, the `fileConfig` passed below will be the one after migration and will not terminate abnormally.

https://github.com/renovatebot/renovate/blob/0d0ab69cfef1098d7b4e5877fcf5f0bce3172056/lib/config-validator.ts#L137
